src/modules/avformat/filter_avfilter.c

```diff
@@ -70,7 +75,8 @@ static void property_changed( mlt_service owner, mlt_filter filter, mlt_event_da
 			{
 				if( !strcmp( opt->name, name + PARAM_PREFIX_LEN ) )
 				{
-					pdata->reset = 1;
+					const char* value = mlt_properties_get(MLT_FILTER_PROPERTIES(filter), name);
+					pdata->reset = non_animated(value);
```

Comment:
I think there is a good opportunity here to skip the reset when the AVOption has the `AV_OPT_FLAG_RUNTIME_PARAM` flag, and to skip setting it below in `set_avfilter_options()` under the sands condition because it will be set in `send_avformat_commands()`.

Reply:
I agree. That would be a good optimization.

Reply:
done